src/Ops.Controllers/Areas/Admin/GroupsController.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Ocuda.Ops.Controllers.Abstract;
+using Ocuda.Ops.Controllers.Areas.Admin.ViewModels.Group;
+using Ocuda.Ops.Controllers.Filters;
+using Ocuda.Ops.Service.Filters;
+using Ocuda.Ops.Service.Interfaces.Ops.Services;
+using Ocuda.Promenade.Models.Entities;
+using Ocuda.Utility.Exceptions;
+using Ocuda.Utility.Keys;
+using Ocuda.Utility.Models;
+
+namespace Ocuda.Ops.Controllers.Areas.Admin
+{
+    [Area("Admin")]
+    [Authorize(Policy = nameof(ClaimType.SiteManager))]
+    [Route("[area]/[controller]")]
+    public class GroupsController: BaseController<GroupsController>
+    {
+        private readonly IGroupService _groupService;
+        private readonly IConfiguration _config;
+
+        public static string Name { get { return "Groups"; } }
+
+        public GroupsController(IConfiguration config,
+            ServiceFacades.Controller<GroupsController> context,
+            IGroupService groupService) : base(context)
+        {
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _groupService = groupService
+                ?? throw new ArgumentNullException(nameof(groupService));
+        }
+
+        [HttpGet("")]
+        [HttpGet("[action]")]
+        public async Task<IActionResult> Index(int page = 1)
+        {
+            var itemsPerPage = await _siteSettingService
+                .GetSettingIntAsync(Models.Keys.SiteSetting.UserInterface.ItemsPerPage);
+
+            var filter = new BaseFilter(page, itemsPerPage);
+
+            var groupList = await _groupService.GetPaginatedListAsync(filter);
+
+            var paginateModel = new PaginateModel
+            {
+                ItemCount = groupList.Count,
+                CurrentPage = page,
+                ItemsPerPage = filter.Take.Value
+            };
+
+            if (paginateModel.PastMaxPage)
+            {
+                return RedirectToRoute(new
+                    {
+                        page = paginateModel.LastPage ?? 1
+                    });
+            }
+
+            return View(new GroupViewModel
+            {
+                AllGroups = groupList.Data,
+                PaginateModel = paginateModel
+            });
+        }
+        [HttpGet("{groupStub}")]
```

Comment:
Insert a blank line between the end of a method and the `[]` attributes of the next.